synapseclient/models/team.py

```diff
@@ -1 +1,357 @@
+    """The ID of the team"""
+
+    name: Optional[str] = None
+    """The name of the team"""
+
+    description: Optional[str] = None
+    """A short description of the team"""
+
+    icon: Optional[str] = None
+    """A file handle ID for the icon image of the team"""
+
+    can_public_join: Optional[bool] = False
+    """True if members can join without an invitation or approval"""
+
+    can_request_membership: Optional[bool] = True
+    """True if users can create a membership request to join"""
+
+    etag: Optional[str] = None
+    """Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates
+        Since the E-Tag changes every time an entity is updated it is used to detect when
+        a client's current representation of an entity is out-of-date."""
+
+    created_on: Optional[str] = None
+    """The date this team was created"""
+
+    modified_on: Optional[str] = None
+    """The date this team was last modified"""
+
+    created_by: Optional[str] = None
+    """The ID of the user that created this team"""
+
+    modified_by: Optional[str] = None
+    """The ID of the user that last modified this team"""
+
+    def fill_from_dict(
+        self, synapse_team: Union[Synapse_Team, Dict[str, str]]
+    ) -> "Team":
+        self.id = (
+            int(synapse_team.get("id", None)) if synapse_team.get("id", None) else None
+        )
+        self.name = synapse_team.get("name", None)
+        self.description = synapse_team.get("description", None)
+        self.icon = synapse_team.get("icon", None)
+        self.can_public_join = synapse_team.get("canPublicJoin", False)
+        self.can_request_membership = synapse_team.get("canRequestMembership", True)
+        self.etag = synapse_team.get("etag", None)
+        self.created_on = synapse_team.get("createdOn", None)
+        self.modified_on = synapse_team.get("modifiedOn", None)
+        self.created_by = synapse_team.get("createdBy", None)
+        self.modified_by = synapse_team.get("modifiedBy", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Team_Create: {self.name}"
+    )
+    async def create(self, synapse_client: Optional[Synapse] = None) -> "Team":
+        """Creates a new team on Synapse.
+
+        Arguments:
+            synapse_client: If not passed in or None this will use the last client from the `.login()` method.
+
+        Returns:
+            Team: The Team object.
+        """
+        loop = asyncio.get_event_loop()
+        current_context = context.get_current()
+        team = await loop.run_in_executor(
+            None,
+            lambda: run_and_attach_otel_context(
+                lambda: Synapse.get_client(synapse_client=synapse_client).create_team(
+                    name=self.name,
+                    description=self.description if self.description else None,
+                    icon=self.icon if self.icon else None,
```

Comment:
Here you aren't passing in the "canPublicJoin" parameter, so it wouldn't be configuring the Team correctly on Synapse.